Fix deploy bugs and rspack config for vsce

t1m0thyj opened the pull request:
**What It Does**
<!-- A list of relevant issues, enhancements, fixed bugs, etc -->
* Fix breakpoints and incremental build not working for SDK package in VS Code debugger ([#55](https://github.com/zowe/zowe-native-proto/issues/55))
* Fix issue found by @jace-roell where SSH profiles with incorrect port may cause deploy to hang
* Fix issue found by @traeok where SSH profiles with empty hostname were shown in quickpick

**How to Test**
<!-- If a bug has been fixed, how can reviewers verify that the change(s) fixed it? -->

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [ ] updated the changelog
- [ ] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)


**Additional Comments**
<!-- Anything else noteworthy about this pull request. This section is optional. -->

t1m0thyj commented:
> When I try to run the `Zowe-SSH: Connect to Host...` command, I get an error during deployment:
> 
> ![image](https://private-user-images.githubusercontent.com/86500639/409273484-aa484471-66dd-4ecf-996b-b53a54766a5d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2MTQ2MDcsIm5iZiI6MTczODYxNDMwNywicGF0aCI6Ii84NjUwMDYzOS80MDkyNzM0ODQtYWE0ODQ0NzEtNjZkZC00ZWNmLTk5NmItYjUzYTU0NzY2YTVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjAzVDIwMjUwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4NDI1Y2UzZWRiNDI2MDQ0OTFkNzE0ZTYwMzg2NGMwNTY5NjhlZTcwZjc0NjAxMzIzZmNiNjI0OGNkODUwMDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ROAELzLcltygf7EyqSKI5r3Fdh-43TdYKcz9cmb4S3U)
> 
> Am I missing any steps? I ran `npm install && npm run build` at the root of the repo and then ran the extension from the Run & Debug panel.

You need to run `npm run tools:artifacts` to download the server PAX artifact from z/OS after building it.

I added a mention of this to the top level readme - let me know if there's somewhere else that would be good to put it 😋

traeok commented:
> @traeok You need to run `npm run tools:artifacts` to download the server PAX artifact from z/OS after building it.
> 
> I added a mention of this to the top level readme - let me know if there's somewhere else that would be good to put it 😋

Do you mean that you're adding this to the top-level readme or that you have added it already? I was following [this section of the README on `main`](https://github.com/zowe/zowe-native-proto?tab=readme-ov-file#deploy--build).

t1m0thyj commented:
> > @traeok You need to run `npm run tools:artifacts` to download the server PAX artifact from z/OS after building it.
> > I added a mention of this to the top level readme - let me know if there's somewhere else that would be good to put it 😋
> 
> Do you mean that you're adding this to the top-level readme or that you have added it already? I was following [this section of the README on `main`](https://github.com/zowe/zowe-native-proto?tab=readme-ov-file#deploy--build) which does not mention running the `tools:artifacts` script

It was mentioned in the section below that one, but I've moved it to be clearer

traeok commented:
> It was mentioned in the section below that one, but I've moved it up to be clearer

I think that makes more sense since its needed for Deploy & Build. Thanks for addressing that, LGTM
